src/Aspire.Hosting.Docker/DockerComposePublishingContext.cs

```diff
@@ -258,4 +260,83 @@ private static void HandleComposeFileVolumes(DockerComposeServiceResource servic
             composeFile.AddVolume(newVolume);
         }
     }
+
+    private void HandleComposeFileBindMounts(DockerComposeServiceResource serviceResource)
+    {
+        foreach (var volume in serviceResource.Volumes.Where(volume => volume.Type == "bind"))
+        {
+            if (string.IsNullOrEmpty(volume.Source))
+            {
+                continue;
+            }
+
+            try
+            {
+                // Determine the destination path for copying the bind mount source
+                var serviceDir = Path.Combine(OutputPath, serviceResource.Name);
+                Directory.CreateDirectory(serviceDir);
+
+                string copiedSourcePath;
+                if (File.Exists(volume.Source))
+                {
+                    // Handle file bind mount
+                    var fileName = Path.GetFileName(volume.Source);
+                    copiedSourcePath = Path.Combine(serviceDir, fileName);
+                    File.Copy(volume.Source, copiedSourcePath, overwrite: true);
+                    
+                    // Update the volume source to use relative path
+                    // Use unix style path separators even on Windows for docker-compose compatibility
+                    volume.Source = Path.GetRelativePath(OutputPath, copiedSourcePath).Replace('\\', '/');
+                }
+                else if (Directory.Exists(volume.Source))
+                {
+                    // Handle directory bind mount
+                    var sourceDirName = Path.GetFileName(volume.Source.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar));
+                    if (string.IsNullOrEmpty(sourceDirName))
+                    {
+                        // If the source is a root path, use a generic folder name
+                        sourceDirName = "data";
+                    }
+                    
+                    copiedSourcePath = Path.Combine(serviceDir, sourceDirName);
+                    CopyDirectory(volume.Source, copiedSourcePath);
+                    
+                    // Update the volume source to use relative path
+                    // Use unix style path separators even on Windows for docker-compose compatibility
+                    volume.Source = Path.GetRelativePath(OutputPath, copiedSourcePath).Replace('\\', '/');
+                }
+                else
+                {
+                    logger.LogWarning("Bind mount source '{Source}' does not exist and will not be copied.", volume.Source);
```

Comment:
These logging messages should use extension methods in DockerComposePublisherLoggerExtensions.cs like the other logging in this file.